### ISSUE TITLE
`empty_structs_with_brackets`:  cannot convert struct if there is a binding with the same name

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#14990]
Using the following flags

`--force-warn clippy::empty-structs-with-brackets`

this code:

```rust
#![allow(non_camel_case_types)]
#![allow(dead_code)]
//#![warn(clippy::empty_structs_with_brackets)]

struct catch {}

pub fn main() {
    let mut catch = true;
    catch = false;
}

```

caused the following diagnostics:

```
    Checking _main v0.1.0 (/tmp/icemaker_global_tempdir.heFzJexA7glb/icemaker_clippyfix_tempdir.TGyi5AnYV0mE/_main)
warning: found empty brackets on struct declaration
 --> src/main.rs:5:13
  |
5 | struct catch {}
  |             ^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#empty_structs_with_brackets
  = note: requested on the command line with `--force-warn clippy::empty-structs-with-brackets`
  = help: remove the brackets

warning: `_main` (bin "_main") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.11s

```

However after applying these diagnostics, the resulting code:

```rust
#![allow(non_camel_case_types)]
#![allow(dead_code)]
//#![warn(clippy::empty_structs_with_brackets)]

struct catch;

pub fn main() {
    let mut catch = true;
    catch = false;
}

```
no longer compiled:

```
    Checking _main v0.1.0 (/tmp/icemaker_global_tempdir.heFzJexA7glb/icemaker_clippyfix_tempdir.TGyi5AnYV0mE/_main)
error[E0530]: let bindings cannot shadow unit structs
 --> src/main.rs:8:13
  |
5 | struct catch;
  | ------------- the unit struct `catch` is defined here
...
8 |     let mut catch = true;
  |             ^^^^^ cannot be named the same as a unit struct

For more information about this error, try `rustc --explain E0530`.
error: could not compile `_main` (bin "_main" test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_main` (bin "_main") due to 1 previous error

```

Version:
```
rustc 1.89.0-nightly (cf423712b 2025-06-05)
binary: rustc
commit-hash: cf423712b9e95e9f6ec84b1ecb3d125e55ac8d56
commit-date: 2025-06-05
host: x86_64-unknown-linux-gnu
release: 1.89.0-nightly
LLVM version: 20.1.5

```

**Comment by samueltardieu** [comment]
I'm torn about this one. Indeed the suggestion causes an error, but I'd say the problem is more in using a lower-case struct name, and using a variable with the same name as a parameterless struct that is in scope.

Note that we cannot simply look at the current crate or module, as the structure name may be imported. Also, even a struct name such as `FooBar` would have the same issue if a variable was created with the same camel cased `FooBar`.

I would be tempted to ignore this issue.
